cosmos/precompile/erc20/erc20.go

```diff
@@ -245,6 +248,38 @@ func (c *Contract) TransferERC20ToCoinTo(
 	return err == nil, err
 }
 
+// PerformBankTransfer transfers ERC20 tokens to SDK coins from msg.sender to recipient.
+func (c *Contract) PerformBankTransfer(
+	ctx context.Context,
+	sender common.Address,
+	recipient common.Address,
+	amount *big.Int,
+) (bool, error) {
+	polarCtx := vm.UnwrapPolarContext(ctx)
+
+	// the caller SHOULD be an ERC20 contract that is registered.
+	tokenHexAddr := polarCtx.MsgSender().Hex()
+
+	// We check to see if the denom exists.
+	if resp, err := c.em.CoinDenomForERC20Address(
+		ctx,
+		&erc20types.CoinDenomForERC20AddressRequest{
+			Token: tokenHexAddr,
+		},
+	); err != nil {
+		// if we error return false
+		return false, err
+	} else if !erc20types.IsPolarisDenom(resp.Denom) {
```

Comment:
if its a polaris denom, its a polaris coin hence erc20 originated. This is the case in which you want to revert with `unauthorized`. We only allow sdk originated coins with polaris erc20s to use this right?
```suggestion
	} else if erc20types.IsPolarisDenom(resp.Denom) {
```

Reply:
yeah so if its NOT a polaris denom we should revert, if it is a polaris denom then it's chill